library/spdm_responder_lib/libspdm_rsp_measurement_extension_log.c

```diff
@@ -0,0 +1,150 @@
+/**
+ *  Copyright Notice:
+ *  Copyright 2024 DMTF. All rights reserved.
+ *  License: BSD 3-Clause License. For full text see link: https://github.com/DMTF/libspdm/blob/main/LICENSE.md
+ **/
+#include "internal/libspdm_responder_lib.h"
+
+#if LIBSPDM_ENABLE_CAPABILITY_MEL_CAP
+
+libspdm_return_t libspdm_get_response_measurement_extension_log(libspdm_context_t *spdm_context,
+                                                                size_t request_size,
+                                                                const void *request,
+                                                                size_t *response_size,
+                                                                void *response)
+{
+    const spdm_get_measurement_extension_log_request_t *spdm_request;
+    spdm_measurement_extension_log_response_t *spdm_response;
+    uint32_t offset;
+    uint32_t length;
+    size_t remainder_length;
+    size_t response_capacity;
+    libspdm_session_info_t *session_info;
+    libspdm_session_state_t session_state;
+    spdm_measurement_extension_log_dmtf_t *spdm_mel;
+    size_t spdm_mel_len;
+
+    spdm_request = request;
+
+    /* -=[Check Parameters Phase]=- */
+    LIBSPDM_ASSERT(spdm_request->header.request_response_code ==
+                   SPDM_GET_MEASUREMENT_EXTENSION_LOG);
+
+    if (libspdm_get_connection_version(spdm_context) < SPDM_MESSAGE_VERSION_13) {
+        return libspdm_generate_error_response(spdm_context,
+                                               SPDM_ERROR_CODE_UNSUPPORTED_REQUEST,
+                                               0,
+                                               response_size, response);
+    }
+
+    if (spdm_request->header.spdm_version != libspdm_get_connection_version(spdm_context)) {
+        return libspdm_generate_error_response(spdm_context,
+                                               SPDM_ERROR_CODE_VERSION_MISMATCH, 0,
+                                               response_size, response);
+    }
+    if (spdm_context->response_state != LIBSPDM_RESPONSE_STATE_NORMAL) {
+        return libspdm_responder_handle_response_state(
+            spdm_context,
+            spdm_request->header.request_response_code,
+            response_size, response);
+    }
+    if (spdm_context->connection_info.connection_state < LIBSPDM_CONNECTION_STATE_NEGOTIATED) {
+        return libspdm_generate_error_response(spdm_context,
+                                               SPDM_ERROR_CODE_UNEXPECTED_REQUEST,
+                                               0, response_size, response);
+    }
+
+    if (spdm_context->connection_info.algorithm.measurement_hash_algo ==
```

Comment:
Filed https://github.com/DMTF/SPDM-WG/issues/3502 for this. If it is illegal then this is an endpoint configuration error and it would need to be caught in `libspdm_check_context`.

Reply:
Thanks.  I have moved the check to `libspdm_check_context`.